### ISSUE TITLE
Seg Fault when repeatedly calling the `edit` Scalar Function

### ISSUE THREAD

**Issue by aarashy** [duckdb/duckdb#3397]
#### What happens?
While trying to understand what certain DuckDB functions do, I couldn't find any information about the `edit` function. So I tried calling it. The first time I did this, the results were very strange, changing from call to call and eventually producing a Seg Fault. 

The below queries should give you a reproducible seg fault. Sometimes I get Seg Fault on the 1st try, sometimes it takes as many as 10, that part is not consistent. If you can invoke it 20 times without getting a seg fault on your environment, that would be interesting to me.

Also, does anybody know what this `edit` function actually is supposed to do?

#### To Reproduce
```
create table test (c1 bigint);
insert into test values (5), (5), (5), (2);
select edit(c1) from test;
select edit(c1) from test;
select edit(c1) from test;
select edit(c1) from test;
Segmentation fault: 11
```

Environment:
OS: Monterey 12.2
DuckDB Version: DuckDB 0.3.2.
DuckDB Client: CLI

**Comment by Mytherin** [comment]
Hah, I only found out about the existence of edit (and other similar) functions today. These functions are UDFs registered by the SQLite shell, see [here](https://www.sqlite.org/cli.html#the_edit_sql_function). Because we support the SQLite API the functions get created into DuckDB, but it seems there is a bug that causes the SQLite UDFs not to work correctly in this case.

**Comment by Mytherin** [comment]
When invoked with the correct parameters it does seem to work:

```sql
select edit('hello world', 'vim');
```

**Comment by aarashy** [comment]
Oh weird, so it's a function that can invoke a text editor like that 😲 

I suppose the action item for this issue would be to correctly throw an error when the edit function is invoked incorrectly?

**Comment by Mytherin** [comment]
Fixed by #3612
